.github/workflows/publish-charts.yaml

```diff
@@ -0,0 +1,89 @@
+name: publish-charts
+
+on:
+  workflow_dispatch:
+    inputs:
+      repo:
+        description: 'Repository from where we want to build Charts'
+        required: true
+      repo_ref:
+        description: 'The branch/tag that will be used as new version for charts. Usually tag ref'
+        required: true
+      repo_chart:
+        description: 'Object with source path, destination path and chart name'
+        required: true
+      currentChartVersion:
+        description: 'Chart version set in `Chart.yaml`. Default: `LATEST`'
+        required: false
+        default: 'LATEST'
+
+env:
+  REPO: ${{ github.event.inputs.repo }}
+  REPO_REF: ${{ github.event.inputs.repo_ref }}
+  currentChartVersion: ${{ github.event.inputs.currentChartVersion }}
+
+jobs:
+  publish-charts:
+    runs-on: ubuntu-latest
+    name: Publish charts
+    continue-on-error: true
+    strategy:
+      max-parallel: 1
+      matrix: ${{ fromJson(github.event.inputs.repo_chart) }}
```

Comment:
@ljubon is it possible to construct this input as yaml, in the caller, than to use JSON, as with yaml is more type-safe?

Reply:
At the moment, best i could find as new solution is to separate theese `files` attributes from workflow and put it in separate file e.g `chart.yaml` which would look something like this
```
{ 
  "REPO": "ljubon/siembol",
  "CurrentChartVersion": "0.1.8",
  "REPO_CHART": {
    "include": [{
      "chart": "siembol",
      "src_path": "deployment/helm-k8s/",
      "dst_path": "siembol/"
    }]
  }
}
```

Task `Publish chart` will be slightly smaller with this, and maybe easier to read
```      
- name: Publish chart
        env: 
          GITHUB_TOKEN: ${{ steps.token.outputs.token }}
          GH_WORKFLOW: "publish-charts.yaml" # workflow to execute 
          GH_CHART_REPO: "ljubon/charts" # owner/repo where charts are published 
          GH_PAGES_BRANCH: "WIP-refactor-gh-pages" # gh-pages
        run: |
          gh workflow run ${{ env.GH_WORKFLOW }} --repo ${{ env.GH_CHART_REPO }} \
          --ref ${{ env.GH_PAGES_BRANCH }} \
          --field repo_ref="${GITHUB_REF##*/}" \
          --json < chart.yaml
```

Reply:
with this approach, i have at the moment problem with `--json < chart.yaml` as it's not parsing it well, bu ti think i could make this work if we go with this approach
let me know what you think :)

Reply:
@ljubon yes, it's much cleaner by moving the contents to a single file. However, i think we can do better than this. `yq` to the rescue again! Check this out: https://mikefarah.gitbook.io/yq/usage/convert. `yq` can convert json to yaml, and yaml to json. With that said, i believe we can actually create a Chart.yaml file, instead of json, and use `yq` to output the file as json and give it to the `gh workflow run` command